### ISSUE TITLE
Writing the project using markdown

### ISSUE THREAD

**Issue by Auslum** [possee-org/d3-js-uarg#16]
Wrote the webpage to be easy readable and investigated about the dropdown feature for the code parts.

**Comment by InessaPawson** [comment]
@Auslum Please document your findings and include a list of references you used researching this topic.

**Comment by Auslum** [comment]
For the visualizations it was proposed not to add the charts as .png images, but to render them when the page is loaded. It was given the following project as reference: 

https://numpy.org/user-survey-2020-details/content/2021/demographics.html

When adding the visualizations in MyST, there is a conflict with the dropdown admonitions when trying to run code:

![Image](https://github.com/user-attachments/assets/8af8b3f0-bda7-4c6a-a3ae-ddb7b09affe7)

Here we can see that MyST only allows two types of cells in a .md file: The Code cell and the Markdown cell.

![Image](https://github.com/user-attachments/assets/7207e8a7-845f-4315-9aa7-ecc3b0854ba9)

The difference is that the Code cell allows us to run the code, while the Markdown cell does not have that option. This means that using the Code cell the code will be executable, making it possible to render the charts when loading the webpage, but the code will be in the page as well, without the option to have it in a dropdown admonition since admonitions are only available in the Markdown cell. For this project it is important that the users do not see all the code, so, admonitions cannot be removed from the project.

So, after making some research, there is evidence that the project took as reference (https://numpy.org/user-survey-2020-details/content/2021/demographics.html) uploads the visualizations as .png images as shown below:

![Image](https://github.com/user-attachments/assets/7913ad6d-4bb5-424c-b2f2-8690a00fed74)

![Image](https://github.com/user-attachments/assets/6f7ef9d4-4387-47b4-a323-6c352d2ac840)

![Image](https://github.com/user-attachments/assets/b54c700b-33aa-4683-a296-73bfb5fcb278)

Also, in the MySTMD.org webpage, in the “Generate and Display Rich Outputs” guide, in the “Static images” section, it says that “Non-interactive images are embedded as PNGs”.

In conclusion, after considering the user’s needs in the project, and the references of previous projects, it will be better to add the visualizations as .png images, this will allow the webpage to show the code in dropdown admonitions.

References: 

https://numpy.org/user-survey-2020-details/content/2021/demographics.html

https://mystmd.org/guide/interactive-notebooks

**Comment by Auslum** [comment]
It was proposed to create admonitions for the code parts in the project. Similar to the project in the following link: 

https://numpy.org/user-survey-2020-details/content/2021/demographics.html

Here, the code parts are shown in the page only if the user clicks on a button that displays those parts as dropdown admonitions. Which is a useful tool for users that want to watch the code, and make the webpage easy to see for users that focus more on the charts and visualizations.

MyST supports admonitions, they are used depending on the desired color for it:

- Blue: {note}, {important} 
- Green: {hint}, {seealso}, {tip}
- Orange: {attention}, {caution}, {warning} 
- Red: {danger}, {error}

Admonitions are added in markdown cells. First, the desired admonition tag is added, followed by the title, and the content is placed below, see the following example:

![Image](https://github.com/user-attachments/assets/ccea8e32-8726-49b8-99f1-2ec70582637e)

![Image](https://github.com/user-attachments/assets/3f12745c-3ecc-4911-86dc-61f5f9c1b2de)

To turn an admonition into a dropdown, add the dropdown class to it. See the example below:

![Image](https://github.com/user-attachments/assets/bcfd0414-4880-465e-a5bf-550e088635b7)

![Image](https://github.com/user-attachments/assets/a33dd760-f443-4f0d-a29c-163177ea6708)

![Image](https://github.com/user-attachments/assets/b48dc8d9-6d47-462a-804d-890b6b6edab8)

For this project, this allows the user to see the code only if he/she wants to, making it easier to visualize the survey results.

References:

- https://mystmd.org/guide/admonitions
